kroxylicious/src/main/java/io/kroxylicious/proxy/internal/ResponseOrderer.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+import io.kroxylicious.proxy.frame.Frame;
+
+/**
+ * While processing a request from a Client, we want to enable custom Protocol
+ * Filters to decide to send a response toward the Client instead of forwarding
```

Comment:
We should try and stick to the upstream/downstream language where possible. 
```suggestion
 * Filters to decide to send a response downstream instead of forwarding
```